Changes

```diff
@@ -185,6 +185,9 @@ Next version (4.05.0):
   by zero.
   (Jeremy Yallop)
 
+- GPR#967, PR#6536: Fix Closure so that overapplication evaluation order
+  matches the bytecode compiler and Flambda.
```

Comment:
The Mantis PR number should be 6136, not 6536.  It's also conventional to list the author and the reporter.